### PR TITLE
categorical recoding plugin

### PR DESCRIPTION
**Pull request by d-callan**
this is one of the first derived vars were likely to need. and since it seems like were about to revisit the conversation around derived vars in earnest, i thought id take a minute and make an attempt to draft a plugin, test my understanding.

top questions currently:
1. does this look remotely right?
2. how am i meant to specify and consume config for a derived var?
3. how do i inform the client of derived variable metadata? (data type, shape, range, etc)

### CODE REVIEW

File: src/main/java/org/veupathdb/service/eda/ms/core/derivedvars/plugin/transforms/CategoricalRecoding.java [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+    if (inputVariables.size() != 1 ||
+        !inputVariables.get(0).getType().equals(APIVariableType.STRING)) {
+      throw new ValidationException(getName() + " categorical recoding accepts only a single variable of type " + APIVariableType.STRING);
+    }
+    _targetColumnName = VariableDef.toDotNotation(inputVariables.get(0));
+  }
+
+  @Override
+  public String getValue(Map<String, String> row) {
+    // dont yet know how to get config here
```
Comment (ryanrdoherty) [inline src/main/java/org/veupathdb/service/eda/ms/core/derivedvars/plugin/transforms/CategoricalRecoding.java:25]: Yeah, need to add some more infrastructure code to wire things together.

**Review by ryanrdoherty — COMMENTED** [review]
I think this should probably stay on a branch for now.  If you really need it for use in a compute plugin, I could fill in the gaps to make it work, but how we use derived vars is still TBD at the moment.

**Comment by d-callan** [comment]
This is NOT needed for a compute or anything. We're about to get the derived vars group going again soon and I was doing a bit of homework on them in advance of that. Very happy to leave it on a branch for a while or even abandon it if it's way off from what we need. This was mostly driven by my own curiosity and a desire to know how much work they are.

**Comment by ryanrdoherty** [comment]
@d-callan Ok to close this PR and delete the branch given the work on phase1-derived-vars?

**Pull request closed without merging**